membersince/membersince.php

```diff
@@ -25,8 +25,11 @@ function membersince_display(&$a, &$b)
 {
 	if (current_theme() == 'frio') {
 		// Works in Frio.
+		$html = '<!DOCTYPE html><html><body>' . $b . '</body></html>';
+
 		$doc = new DOMDocument();
-		$doc->loadHTML(mb_convert_encoding($b, 'HTML-ENTITIES', 'UTF-8'));
+		$doc->validateOnParse = true;
+		@$doc->loadHTML(mb_convert_encoding($html, 'HTML-ENTITIES', 'UTF-8'));
```

Comment:
I really don't like this error suppression operator, we should fix what is triggering errors instead.

What about the output of `saveHtml` in regard with the final output, are there two `<html>` tags?

Reply:
in the output of `$b` the `<html>` was still there. In the browser output not.

Problem was that on my productive instance the profile page wasn't working because `$elm = $doc->getElementById('aprofile-fullname');` was `null` because of an invalid html document.

I would like to pass this issue to another developer since I have no idea about php html parsing. I have no idea how to get an valid html document so that `loadHTML()` would be silent.

Reply:
Like I said, the easiest solution probably would be to integrate membersince in the core profile template.

Reply:
Yes, I agree with you. But I don't know if everyone would be happy with this solution. I don't want to have another user settings slider and I don't know if anyone does have concerns about privacy.

Reply:
The addional features settings already is a sort of core addon list, you’re just moving the setting from one part of the project to another.

Reply:
As long as it remains an opt-in setting, I don't see a problem. But I don't like the general trend to include more and more addons into the core.

Reply:
Maybe I find some time in the evening to port it to core.

The problem is that some addons are rather hackish solutions. Especially if there is no good interface to the core. This is mostly the situation when an addon have to modify an already existent template. The smiley button addon is also such a case

Reply:
@tobiasd This is because there are limited ways in which addons can affect page display.

Beyond that, I fail to see the difference between a small display feature we can add/hide from the additional feature settings from a small display feature we can add/hide from the addons settings, since we recommend to import all the addons anyway.

Reply:
done https://github.com/friendica/friendica/pull/4432

So I close this PR